File: portfolio/src/main/java/com/measqa/rest/ArticleService.java
package com.measqa.rest;

import javax.ws.rs.GET;
import javax.ws.rs.Path;
import javax.ws.rs.PathParam;
import javax.ws.rs.Produces;
import javax.ws.rs.core.MediaType;

import com.measqa.entity.Article;

@Path("/articles")
public class ArticleService extends BaseService {

	

	@GET
	@Path("/{id}")
	@Produces(MediaType.APPLICATION_JSON)
	public Article getArticle(@PathParam("id") int id) {
		return factory.getArticleDao().getArticle(id);

	}
}

File: FrontEnd/app/article.service.ts
import { Injectable }    from '@angular/core';
import { Headers, Http } from '@angular/http';

import 'rxjs/add/operator/toPromise';

import { Article } from './models/article';

@Injectable()
export class ArticleService {

  private headers = new Headers({'Content-Type': 'application/json'});
  private articleUrl = 'http://measqa.com:8081/rest/articles'; 

  constructor(private http: Http) { 
    this.http=http
  }

  getArticle(id: number): Promise<Article> {
    const url = `${this.articleUrl}/${id}`;
	
  let article: Article;
	
	console.log(this.http.get(url));
  console.log(this.http.get(url).toPromise().then(response => response.json() as Article));
	
  return this.http.get(url).toPromise().then(response => response.json() as Article);
}

};
